pfVerticalNavigation: on mobile, move the masthead utility items to the vertical navigation

dtaylor113 opened the pull request:
## Description
Added a `mobileOnly` property to nav. item objects; when 'true' items only display in mobile mode. 
Addresses: #745 

Ex: The **Help** and **User** vert. nav items only appear when in mobile (< 768px) mode, and are hidden when not in mobile mode:
![mobileonly](https://user-images.githubusercontent.com/12733153/41430999-96a34efc-6fdf-11e8-98a9-54bfaa9fc0da.gif)

Note the masthead utility items (Help & User dropdowns) continue to be hidden at mobile; this PR allows app devs to mimic them 'moving' to the vert nav menu.

@alexkieling

## PR Checklist
- [x] Screenshots are attached (if there are visual changes in the UI)
- [x] A Designer (@beanh66) is assigned as a reviewer (if there are visual changes in the UI)

jeff-phillips-18 commented on src/navigation/examples/vertical-navigation-basic.js:
```diff
@@ -392,9 +394,19 @@
+           iconClass: "fa pficon-help",
+           href: "#/help",
+           mobileOnly: "true"
+        },
```
This should have Help and About as children to be consistent with the header.

jeff-phillips-18 replied:
These changes should be made in the `-router` example as well

dtaylor113 replied:
> This should have Help and About as children to be consistent with the header.

Wasn't crazy about having sub-menus:
```
Help -> Help
        About
```
So left out 'About' as maybe something you wouldn't need at mobile.

jeff-phillips-18 replied:
OK, just thinking we should show consistency.

jeff-phillips-18 commented on src/navigation/vertical-navigation.html:
```diff
@@ -30,7 +30,7 @@
                     'show-mobile-nav': $ctrl.showMobileNav}">
       <ul class="list-group">
-        <li ng-repeat="item in $ctrl.items" class="list-group-item"
+        <li ng-repeat="item in $ctrl.items | navItems:$ctrl.inMobileState" class="list-group-item"
```
Might be easer just to add:

```
'visible-xs': item.mobileOnly
```

to the `ng-class` rather than filtering.

dtaylor113 replied:
but is that acceptable performance wise?  Seems a waste to pull in/construct all that dom/html only to not display it.  Seems like it would be faster just filtering items.

jeff-phillips-18 commented on src/navigation/examples/vertical-navigation-basic.js:
```diff
@@ -392,9 +394,19 @@
+           iconClass: "fa pficon-user",
+           mobileOnly: "true",
+           children: [
+              { title: "Preferences" },
```
Can we change `Preferences` to `User Preferences` both here and in the header area? It was a UX request in the patternfly-react repo, so might as well update here as well.

jeff-phillips-18 commented on src/navigation/examples/vertical-navigation-basic.js:
```diff
@@ -35,6 +35,8 @@
  *   <li>.tooltip      - (string) Tooltip to display for the badge
  *   <li>.badgeClass:  - (string) Additional class(es) to add to the badge container
  *   </ul>
+ * <li>.mobileOnly     - (string) When set to 'true', menu item will only be displayed when browser is in mobile mode (<768px).
```
why use a string here rather than a boolean?

dtaylor113 commented:
Hi @jeff-phillips-18, I believe I have addressed your comments, except the one about the Help-at-mobile menu, IMO I think leaving off the 'About' menu item represents a truer @mobile menu.

dtaylor113 commented:
> Might want to address #746 while you are working here...

I'd prefer to do that in a follow up pr, thanks :-)
